packages/angular-17/projects/components/src/lib/directives/tds-dropdown-value-accessor.ts

```diff
@@ -0,0 +1,24 @@
+import { Directive } from '@angular/core';
+import { NG_VALUE_ACCESSOR } from '@angular/forms';
+
+import { ValueAccessor } from '../stencil-generated/value-accessor';
+
+/**
+ * A custom value accessor for the tds-dropdown. It extends the default ValueAccessor that is auto-generated by Stencil
+ * The tds-dropdown requires slightly different event handling since it doesn't act like any of the native inputs.
+ * The main difference between this value accessor and the other value accessors that are auto-generated by Stencil, is that this one handles `$event.detail.value` instead of `$event.target.value`.
+ */
+@Directive({
+  selector: 'tds-dropdown',
+  host: {
+    '(tdsChange)': 'handleChangeEvent($event.detail.value)',
+  },
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: TdsDropdownValueAccessor,
+      multi: true,
+    },
+  ],
+})
+export class TdsDropdownValueAccessor extends ValueAccessor {}
```

Comment:
@adarean5 I was getting warnings on my local with 
`Useless constructor  @typescript-eslint/no-useless-constructor`
so I dropped the constructor here. Please see if it was a false warning. 
The same goes for another directive file.

Reply:
I don't think that the constructor call was necessary, since it only called the super method, and parent constructor does not contain any logic. I made the custom classes as similar as possible to the ones generated by stencil, but I think this should be fine.